### PR TITLE
Adapt httpboot plugin to respond with client specific UKIs

### PR DESCRIPTION
**Pull request by hardikdr**
# Proposed Changes

```
# For static behavior, same UKI for all requests:
- httpboot: <StaticURL>

# For dynamic behavior with bootservice:
- httpboot: bootservice:<BootServiceURL>
```
Fixes #

### CODE REVIEW

**Comment by hardikdr** [comment]
cc @damyan @5kt

File: plugins/httpboot/plugin.go [hunk 1]
```diff
@@ -54,9 +69,26 @@ func setup4(args ...string) (handler.Handler4, error) {
...
+	} else {
+		clientIPs, err := extractClientIP6(req)
+		if err != nil {
+			log.Errorf("failed to extract ClientIP, Error: %v Request: %v ", err, req)
+			return resp, true
+		}
+		ukiURL, err = fetchUKIURL(bootFile6, clientIPs)
+		if err != nil {
+			log.Errorf("failed to fetch UKI URL: %v", err)
+			return resp, true
```
Comment (damyan) [inline plugins/httpboot/plugin.go:85]: Same

File: plugins/httpboot/plugin.go [hunk 2]
```diff
@@ -54,9 +69,26 @@ func setup4(args ...string) (handler.Handler4, error) {
...
 	log.Debugf("Received DHCPv6 request: %s", req.Summary())
+
+	var ukiURL string
+	if !useBootService {
+		ukiURL = bootFile6
+	} else {
+		clientIPs, err := extractClientIP6(req)
+		if err != nil {
+			log.Errorf("failed to extract ClientIP, Error: %v Request: %v ", err, req)
+			return resp, true
```
Comment (damyan) [inline plugins/httpboot/plugin.go:80]: Change to false, so if your plugin fails for whatever reason, the plugin chain will not break (meaning: the next plugin in the configuration will get executed)

File: plugins/httpboot/plugin.go [hunk 3]
```diff
@@ -84,12 +116,19 @@ func Handler6(req, resp dhcpv6.DHCPv6) (dhcpv6.DHCPv6, bool) {
 
 func Handler4(req, resp *dhcpv4.DHCPv4) (*dhcpv4.DHCPv4, bool) {
 	log.Debugf("Received DHCPv4 request: %s", req.Summary())
+
+	ukiURL, err := fetchUKIURL(bootFile4, []string{req.ClientIPAddr.String()})
+	if err != nil {
+		log.Errorf("failed to fetch UKI URL: %v", err)
+		return resp, true
```
Comment (damyan) [inline plugins/httpboot/plugin.go:123]: Same

**Review by damyan — CHANGES_REQUESTED** [review]

File: plugins/httpboot/plugin.go (hunk 1 above)
Comment (hardikdr) [reply plugins/httpboot/plugin.go]: Done

File: plugins/httpboot/plugin.go (hunk 3 above)
Comment (hardikdr) [reply plugins/httpboot/plugin.go]: Done

**Review by 5kt — APPROVED** [review]

**Review by damyan — APPROVED** [review]

**Pull request merged**